From 3c8a3a0b19fcbdb0406199bfdd58afee0f314adb Mon Sep 17 00:00:00 2001
From: Juan Gutierrez <jgutierrez@ti.com>
Date: Thu, 30 Jun 2011 23:52:03 -0500
Subject: [PATCH] OMAP4: HDMI: Disablle DSS clock only if HDMI is disconnected.

This is to avoid synchronization issues when activating/deactivating
the ACR WA.

Signed-off-by: Juan Gutierrez <jgutierrez@ti.com>
Signed-off-by: Ricardo Neri <ricardo.neri@ti.com>
---
 drivers/video/omap2/dss/hdmi.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

Index: kernel/drivers/video/omap2/dss/hdmi.c
===================================================================
--- kernel.orig/drivers/video/omap2/dss/hdmi.c	2011-07-03 15:28:49.679751518 +0530
+++ kernel/drivers/video/omap2/dss/hdmi.c	2011-07-03 15:30:00.572252380 +0530
@@ -1832,8 +1832,9 @@
 
 	hdmi_enable_clocks(0);
 
-	/* cut clock(s) */
-	dss_mainclk_state_disable(true);
+	/* cut clock(s) only if disconnected */
+	if(!hdmi_connected)
+		dss_mainclk_state_disable(true);
 
 	/* reset to default */
 }
